internal/app/app.go

```diff
@@ -41,7 +46,22 @@ func Start() error {
 		return fmt.Errorf("storage error: %v", err)
 	}
 
-	s := server.NewServer(repo, tokenManager, st)
+	producer, err := queue.NewProducer(&conf)
+	if err != nil {
+		return fmt.Errorf("can't create publisher: %w", err)
+	}
+
+	consumer, err := queue.NewConsumer(repo, st, &conf)
+	if err != nil {
+		return fmt.Errorf("can't create consumer: %w", err)
+	}
+
+	err = consumer.ListenToQueue()
+	if err != nil {
+		return fmt.Errorf("queue consumer error: %w", err)
+	}
+
+	s := server.NewServer(repo, tokenManager, st, producer)
```

Comment:
reorder. Put `producer` next to where it's being used

Reply:
> reorder. Put `producer` next to where it's being used

I didn't quite understand what you meant.

Reply:
doesn't matter anymore, but what I meant was:
you had 2 separate things being setup/created here:
1. consumer
2. producer + server
the `producer` is being created on line 49, then you have consumer-related calls, that are totally unrelated to producer-server, and only then on line 64 `producer` is used to create a server.